pkg/cmd/openshift-tests/collect-disk-certificates/collector.go

```diff
@@ -0,0 +1,217 @@
+					os.Exit(0)
+				}
+			}()
+			signal.Notify(abortCh, syscall.SIGINT, syscall.SIGTERM)
+
+			if err := f.Validate(); err != nil {
+				return err
+			}
+
+			o, err := f.ToOptions()
+			if err != nil {
+				return err
+			}
+
+			return o.Run(ctx)
+		},
+	}
+
+	f.AddFlags(cmd.Flags())
+
+	return cmd
+}
+
+func (f *RunCollectDiskCertificatesFlags) AddFlags(flags *pflag.FlagSet) {
+	flags.StringArrayVar(&f.CollectDirs, "collect-dir", f.CollectDirs, "directories to collect certs in")
+	flags.StringVar(&f.Prefix, "prefix", f.Prefix, "directories prefix")
+	f.ConfigFlags.AddFlags(flags)
+	f.OutputFlags.BindFlags(flags)
+}
+
+func (f *RunCollectDiskCertificatesFlags) SetIOStreams(streams genericclioptions.IOStreams) {
+	f.IOStreams = streams
+}
+
+func (f *RunCollectDiskCertificatesFlags) Validate() error {
+	if len(f.OutputFlags.OutFile) == 0 {
+		return fmt.Errorf("output-file must be specified")
+	}
+	if len(f.CollectDirs) == 0 {
+		return fmt.Errorf("dirs must be specified")
+	}
+
+	return nil
+}
+
+func (f *RunCollectDiskCertificatesFlags) ToOptions() (*RunCollectDiskCertificatesOptions, error) {
+	originalOutStream := f.IOStreams.Out
+	closeFn, err := f.OutputFlags.ConfigureIOStreams(f.IOStreams, f)
+	if err != nil {
+		return nil, err
+	}
+
+	restConfig, err := f.ConfigFlags.ToRESTConfig()
+	if err != nil {
+		return nil, err
+	}
+	kubeClient, err := kubernetes.NewForConfig(restConfig)
+	if err != nil {
+		return nil, err
+	}
+
+	return &RunCollectDiskCertificatesOptions{
+		KubeClient:      kubeClient,
+		OutputFile:      f.OutputFlags.OutFile,
+		CloseFn:         closeFn,
+		OriginalOutFile: originalOutStream,
+		IOStreams:       f.IOStreams,
+		CollectDirs:     f.CollectDirs,
+		Prefix:          f.Prefix,
+	}, nil
+}
+
+// RunCollectDiskCertificatesOptions sets options for api server disruption monitor
+type RunCollectDiskCertificatesOptions struct {
+	KubeClient kubernetes.Interface
+	OutputFile string
+
+	CollectDirs []string
+	Prefix      string
+
+	OriginalOutFile io.Writer
+	CloseFn         iooptions.CloseFunc
+	genericclioptions.IOStreams
+}
+
+func (o *RunCollectDiskCertificatesOptions) Run(ctx context.Context) error {
+	ctx, cancelFn := context.WithCancel(ctx)
+	defer cancelFn()
+
+	pkiList := &certgraphapi.PKIList{}
+	errs := []error{}
+
+	restConfig, err := clusterinfo.GetMonitorRESTConfig()
+	if err != nil {
+		return err
+	}
+	kubeClient, err := kubernetes.NewForConfig(restConfig)
+	if err != nil {
+		return err
+	}
+	controlPlaneLabel := labels.SelectorFromSet(map[string]string{"node-role.kubernetes.io/control-plane": ""})
+
+	var nodeList *corev1.NodeList
+	if err := wait.PollImmediate(time.Second, 60*time.Second, func() (bool, error) {
+		var listErr error
+		nodeList, listErr = kubeClient.CoreV1().Nodes().List(ctx, metav1.ListOptions{LabelSelector: controlPlaneLabel.String()})
+		if err != nil {
+			return false, listErr
+		}
+		return true, nil
+	}); err != nil {
+		return fmt.Errorf("failed to list nodes: %v", err)
+	}
```

Comment:
```suggestion
		nodeList, err := kubeClient.CoreV1().Nodes().List(ctx, metav1.ListOptions{LabelSelector: controlPlaneLabel.String()})
 if err != nil{
```

is sufficient.

Reply:
I'm not sure if that would work 100% of the time. In some cases CRB didn't get applied in time, so pod failed to start - and since its not backed with a deployment it went to Error and test failed. In actual tests the nodes will probably take time pulling the image, but I'd rather have this retry just in case

Reply:
what is CRB?

Reply:
ClusterRoleBinding to allow the container to list nodes. Removed retry, lets see if this happens in CI anyway